index.bs

```diff
@@ -49,198 +49,207 @@ conversion domain.
     [=url/origin=].
 
-<h3 algorithm id="creating-impression">Activating an impression</h3>
+<h3 algorithm id="obtaining-attribution-source-anchor">Obtaining an attribution source from an <code>a</code> element</h3>
 
-To <dfn>activate an impression</dfn> from an <{a}> tag |anchor|,
+To <dfn>obtain an attribution source</dfn> from an <{a}> element |anchor|:
 1. Let |currentTime| be the current time.
-1. Let |impression| be a new [=impression=] struct whose items are:
+1. Let |source| be a new [=attribution source=] struct whose items are:
 
-    : [=impression/impression source=]
+    : [=attribution source/source origin=]
     :: |anchor|'s [=relevant settings object=]'s [=environment/top-level origin=].
-    : [=impression/impression data=]
-    :: The result of applying [=parsing conversion data=] to |anchor|'s
-        <{a/impressiondata}> attribute.
-    : [=impression/conversion destination=]
-    :: The result of running [=parse a conversion destination=] on |anchor|.
-    : [=impression/reporting endpoint=]
-    :: The [=url/origin=] of the result of running the [=URL parser=] on the value
-        of |anchor|'s <{a/reportingorigin}> attribute.
-    : [=impression/expiry=]
-    :: |currentTime| +  <{a/impressionexpiry}> milliseconds.
-    : [=impression/impression time=]
+    : [=attribution source/event id=]
+    :: The result of running [=parse attribution data=] with |anchor|'s
+        <{a/attributionsourceeventid}> attribute and [=max event id value=].
+    : [=attribution source/attribution destination=]
+    :: The result of running [=parse an attribution destination=] with |anchor|'s <{a/attributiondestination}> attribute.
+    : [=attribution source/reporting endpoint=]
+    :: The [=url/origin=] of the result of running the [=URL parser=] with
+        |anchor|'s <{a/attributionreportto}> attribute.
+    : [=attribution source/expiry=]
+    :: |currentTime| +  <{a/attributionexpiry}> milliseconds.
+    : [=attribution source/source time=]
     :: |currentTime|.
+1. Return |source|
+
+<dfn>Max event id value</dfn> is a vendor specific integer which controls the meximum size value which can be used as an [=attribution source/event id=]
 
-1. Issue: Need to spec how to store the impression.
+1. Issue: Need to spec how to store the attribution source.
 
-<h3 algorithm id="creating-a-conversion">Creating a conversion</h3>
+<h3 algorithm id="attribution-trigger-creation">Creating an attribution trigger</h3>
 
-To <dfn>create a conversion</dfn> from a [=url=] |url| and an
-[=environment settings object=] |environment|, return a new [=conversion=]
-struct with the items:
+To <dfn>obtain an attribution trigger</dfn> given a [=url=] |url|, and an
+[=environment settings object=] |environment|, return a [=attribution trigger=]
+with the items:
 
-    : [=conversion/conversion source=]
+    : [=attribution trigger/trigger origin=]
     :: |environment|'s [=environment/top-level origin=].
```

Comment:
Why is this the top-level origin instead of just the origin? Can attribution triggers happen in cross-origin iframes? Have you discussed somewhere the security implications of writing a different origin into the attribution trigger, and what it means for specs that need to trust or distrust that value?

Reply:
I was worried about a hostile iframe being able to create somehow-hostile attributions to the surrounding page, but I'm not sure what attacks that would actually be useful for. It helps that the user has to click in order to make the attribution source happen. Maybe hostile triggers are more worrying, since they might cause payment from the trigger to the source? In any case, this sort of attribution (no pun intended) of one origin's actions to a different origin is worth talking about in the security considerations.

Reply:
Yes that would cause issues. Sources are also problematic: a hostile frame logging additional sources could cause credits to be assigned incorrectly, or cause the browser to delete already existing sources in some cases.

The main mitigation we have is [requiring a permissions policy](https://github.com/WICG/conversion-measurement-api#publisher-side-controls-for-attribution-source-declaration) to invoke the API inside of iframes, which is disabled by default for cross-origin iframes.

I will make a note to address this in Security Considerations when the Permissions Policy is added to this draft spec.